runtime/jvmti/jvmtiHelpers.c

```diff
@@ -1601,3 +1659,144 @@ ensureHeapWalkable(J9VMThread *currentThread)
 		}
 	}
 }
+
+#if JAVA_SPEC_VERSION >= 19
+static void
+tlsNullFinalizer(void *entry)
+{
+	/* do nothing */
+}
+#endif /* JAVA_SPEC_VERSION >= 19 */
+
+/**
+ * Allocate a thread local storage key (which represents a jvmtiEnv) to index into a thread
+ * object's TLS array.
+ *
+ * @param[in] vm java vm
+ * @param[out] handle on successful return contains the allocated key value
+ * @return 0 on success, -1 on failure
+ */
+IDATA
+jvmtiTLSAlloc(J9JavaVM *vm, UDATA *handle)
+{
+#if JAVA_SPEC_VERSION < 19
+	return omrthread_tls_alloc(handle);
+#else /* JAVA_SPEC_VERSION < 19 */
+	return jvmtiTLSAllocWithFinalizer(vm, handle, tlsNullFinalizer);
+#endif /* JAVA_SPEC_VERSION < 19 */
```

Comment:
It's better to follow one pattern `>= 19` (preferred) vs. `< 19`. Preferred since it keeps the newest version on top.
```suggestion
#if JAVA_SPEC_VERSION >= 19
	return jvmtiTLSAllocWithFinalizer(vm, handle, tlsNullFinalizer);
#else /* JAVA_SPEC_VERSION >= 19 */
	return omrthread_tls_alloc(handle);
#endif /* JAVA_SPEC_VERSION >= 19 */
```